fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpapi/applicationpasswords/ApplicationPasswordNetwork.kt

```
@@ -103,6 +103,10 @@ class ApplicationPasswordNetwork @Inject constructor(
         }
     }
 
+    suspend fun deleteApplicationPassword(site: SiteModel) {
+        applicationPasswordManager.deleteApplicationCredentials(site)
```

Do we not need to pass the return value from `deleteApplicationCredentials()`?

---

We agreed on making the call fail silently during the logout, but I agree it should still return something.
For not, this function is just temporary, I added it because the `ApplicationPasswordsManager` is internal, but I don't like it that much, I don't think the `Network` class should be the one responsible for this, it would be better to be part of `AccountStore` or `SiteStore`.

I will try to move it there, I already tried and was blocked by the fact that injecting `ApplicationPasswordsManager` in any one of those classes will require all of the FluxC clients to break until they provide an `@ApplicationPasswordsClientId` String, which I don't want to do, as the feature should be kept optional until it's needed.

Let's keep this way in this PR, I'll see what I can do in another PR, WDYT?

---

@hafizrahman I handled this change in this PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2600, I'll update the description this afternoon, anyway, it's not urgent, feel free to leave it until next week.

---

Sounds like a good plan to me!

---

> I don't think the Network class should be the one responsible for this, it would be better to be part of AccountStore or SiteStore.

It did feel out of place for me but I was struggling to articulate it properly. 

> will require all of the FluxC clients to break until they provide an @ApplicationPasswordsClientId String, which I don't want to do, as the feature should be kept optional until it's needed.

Interesting issue, I saw #2600 and so far couldn't think of a more simple solution for this. Almost makes me want to keep this in this class just to keep things simple. Calling password deletion manually doesn't feel like a common use case to me for a FluxC client, so perhaps it's OK to keep it tucked here?

---

> Calling password deletion manually doesn't feel like a common use case to me for a FluxC client

We will be calling it on logout, so it's needed.

> perhaps it's OK to keep it tucked here?

Can you elaborate? you don't like the solution done in #2600? I tried to solve the original issue with it, the feature is still optional, and even without providing `@ApplicationPasswordsClientId`, the clients will keep working, things would break only when they start using the feature itself at runtime.